OutOfSchool/OutOfSchool.AuthServer.Tests/Services/Password/PasswordGeneratorTests.cs

```diff
@@ -0,0 +1,19 @@
+﻿using NUnit.Framework;
+using OutOfSchool.AuthCommon.Services.Password;
+
+namespace OutOfSchool.AuthServer.Tests.Services.Password;
+
+[TestFixture]
+public class PasswordGeneratorTests
+{
+    [Test]
+    public void PasswordGenerator_ShouldReturnValidPassword()
+    {
+        // Act
+        var result = PasswordGenerator.GenerateRandomPassword();
+
+        // Assert
+        Assert.IsNotNull(result);
+        Assert.That(result, Does.Match(@"^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8}$"));
```

Comment:
2 issues:
(Readability) This regexp is as cool as complicated. Unit test code should be simple. 

Please consider writing instead of 1 complicated 6 simple asserts: 4 checking that characters from each charset are present in the password, and 1 checking that no other characters are present, and one checking password length. 
It will improve readablility and make subsequent changes easier (just imagine adding 5th checked character set, or changing the existing one).

(Logic) Is it strict requirement that generated password length should be `8`? Or should it be `>= 8`? 
If latter, please change the code.

Reply:
The length of the generated password is equal to the PasswordMinLength constant.
Fixed. Thank you!

Reply:
Thanks for fixing. One last touch of perfection could be made if you add message in your `Assert`s. To get better explanation for people reading the test code and failed test message.

Like 
```
Assert.IsFalse(result.Any(c => !char.IsLetterOrDigit(c) && !Constants.ValidationSymbols.Contains(c)), 
    "Password shouldn't contain any other symbols than digits, upper/lowercase letters, and allowed special symbols");
```

Reply:
I've added message. Thank you!